kernel/aster-nix/src/syscall/socket.rs

```diff
@@ -35,7 +35,7 @@ pub fn sys_socket(domain: i32, type_: i32, protocol: i32) -> Result<SyscallRetur
         (
             CSocketAddrFamily::AF_INET,
             SockType::SOCK_DGRAM,
-            Protocol::IPPROTO_IP | Protocol::IPPROTO_UDP,
+            Protocol::IPPROTO_IP | Protocol::IPPROTO_UDP | Protocol::IPPROTO_ICMP,
```

Comment:
This change is unrelated to the PR and should be reverted.

This PR does not properly implement the `IPPROTO_ICMP` protocol, so there is no point in changing this line.

Reply:
Agreed, this is left unchecked, revert changes!